File: typecast-ruby/examples/with_timestamps.rb
require "typecast"

client = Typecast::Client.new
response = client.text_to_speech_with_timestamps(
  Typecast::Models::TTSRequest.new(
    voice_id: "tc_60e5426de8b95f1d3000d7b5",
    text: "Hello with timestamps.",
    model: Typecast::Models::TTS_MODEL_V30,
    language: "eng"
  ),
  granularity: "word"
)

response.save_audio("hello.wav")
puts response.to_srt

File: typecast-dart/example/with_timestamps.dart
import 'package:typecast_dart/typecast.dart';

Future<void> main() async {
  final client = TypecastClient();
  final response = await client.textToSpeechWithTimestamps(
    const TtsRequest(
      voiceId: 'tc_60e5426de8b95f1d3000d7b5',
      text: 'Hello with timestamps.',
      model: TtsModel.ssfmV30,
      language: LanguageCode.eng,
    ),
    granularity: 'word',
  );
  await response.saveAudio('hello.wav');
  print(response.toSrt());
}
